lib/rpc.ts

```diff
@@ -76,3 +120,90 @@ export async function createSenderLink(connection: any, senderOptions: SenderOpt
     sender: sender
   };
 }
+
+export function sendRequest(connection: any, link: RequestResponseLink, request: AmqpMessage): Promise<any> {
+  if (!connection) {
+    throw new Error("connection is a required parameter and must be of type 'object'.");
+  }
+
+  if (!link) {
+    throw new Error("link is a required parameter and must be of type 'object'.");
+  }
+
+  if (!request) {
+    throw new Error("request is a required parameter and must be of type 'object'.");
+  }
+  return new Promise((resolve: any, reject: any) => {
+    // TODO: Handle timeout incase SB/EH does not send a response.
+    const messageCallback = (context: Context) => {
```

Comment:
>messageCallback [](start = 10, length = 15)

it does not look like you're matching request/responses with a correlationId. this assumes everything is always in order and nothing ever fails. What if you disconnect right after sending a request, then reconnect and send another request? the first message you receive might be the response from the previous message right? I think it'd be safer to match correlationIds and just release messages whose correlationId you don't care for.

Reply:
That is true. Can make the change. To make sure I understood this, Can you validate the following:

- I shall be adding a correlation_id to the request if the caller to `sendRequest` did not provide one. 
- When the response is received, I shall match the correlationId in the request with the one present in the response. 
- If that matches then 
  - the Promise is resolved else 
  - the Promise is rejected or not ?? If not rejected then when will the promise be rejected (apart from error conditions)?

Reply:
you'd have to double check how correlationIds are handled with the Event hubs team. it might be that the messageId is sent in the request and is returned as a correlationId in the response, i'm not sure.
If it matches and status code is 200, you resolves, it it matches with an error you reject, and if it doesn't match you release the message and keep receiving until you receive the one with the right correlationId (or time out)

---
In reply to: [182284231](https://github.com/Azure/azure-event-hubs-node/pull/49#discussion_r182284231) [](ancestors = 182284231)